.github/workflows/benchmark_asterinas.yml

```diff
@@ -41,7 +41,9 @@ jobs:
         command: |
           make install_osdk
           bash test/benchmark/bench_linux_and_aster.sh ${{ matrix.benchmark }}
-        on_retry_command: make clean
+        on_retry_command: |
+          make clean
+          make --no-print-directory -C test/apps/mongoose clean
```

Comment:
First of all, I don't think `make clean` is desirable. Compiling the kernel is really time consuming. Having to `make clean` on retry means we are going to repeat a lot of efforts. And it means our Makefile is not written in a robust way. Ideally, if anything bad happens during `make install_osdk` or `bench_linux_and_aster.sh`. We should be able to simply rerun `make install_osdk` or `bench_linux_and_aster.sh`.

Second, I don't know why `mongoose` test has to be so special that the benchmark CI code has to single it out. The benchmark-related code should not be concerned with anything particular tests. If the `mongoose` test cannot be re-run on retry, then it means the `mongoose` test is poorly written and should be fixed.

Reply:
Our makefile is indeed not perfect and lacks a logic to verify whether the download is successful. For example, when `vdso64.so` was downloaded in real time before, `vdso64.so` would be downloaded through wget every time it was compiled. If this compilation fails by accident, the makefile will not re-download it next time because the `vdso64.so` file exists, causing subsequent compilation failures.

I hope that no residual files will be left before the download process is completed, so as not to affect the judgment of the makefile. Therefore, adding a temporary file is more reasonable.
```makefile
$(MONGOOSE_FILES): | $(MONGOOSE_DIR)
@set -e;
wget -c -O $@.tmp "https://raw.githubusercontent.com/cesanta/mongoose/7.13/$(notdir $@)"
@mv $@.tmp $@
```
The above code will not generate `$(MONGOOSE_FILES)` if the download process times out or is unexpectedly interrupted.